### ISSUE TITLE
Permission "Mount" to restrict users to specific entries in collections

### ISSUE THREAD

**Issue by flolanger** [statamic/ideas#1216]
Currently, it is not possible to restrict a CP user to specific entries of collections. For example, if you have a "pages" collection with many entries, all users with access to that collection will be able to view and edit all entries.

That's fine for small sites, but if you have a larger site with multiple editors, administrators may want to restrict users to specific areas. In other CMS, this can be done with user "mounts". You can simply select one or more entries that the user has access to, and even allow subentries. For example, if an editor is only responsible for updating the "about us" page with the "team" child page, an administrator can simply allow the user to view and edit "about us" and its child entries.

Other entries are completely hidden from the user, which also improves the UX when there are many pages/entries.

Let me know what you think. I would also love to hear about other solutions from people who had the same issue.

**Comment by j3ll3yfi5h** [comment]
@flolanger Did you already have a look on the author permissions?

https://statamic.dev/users#author-permissions

**Comment by flolanger** [comment]
> @flolanger Did you already have a look on the author permissions?
> 
> https://statamic.dev/users#author-permissions

@j3ll3yfi5h Yes, but that doesn't provide quite the feature I would want. It should be controlled on the roles / permissions level and not on entry level. Also, there may be multiple roles that should be able to edit certain areas on the site.

**Comment by duncanmcclean** [comment]
You might be able to achieve part of this right now with some custom code...

For example: to limit which entries are visible in the collection listing table (not the tree view), you could take advantage of the [`EntriesIndexQuery` hook](https://statamic.dev/extending/hooks#entry-index-query-query).

Then you could override Statamic's `EntryPolicy` and do any of your own checks in there too.
